src/stores/theme/createThemeStore.tsx

```diff
@@ -1,67 +1,213 @@
+        isBefore(new Date(), themeState.expirationDate),
+    )
+    .map(([uid]) => uid);
+
+  return state.partnerThemes.filter((theme) =>
+    availableUids.some((uid) => uid === theme.uid),
+  );
+};
+
+const CONFIG_THEME_EXPIRATION_DAYS = 7;
+
+const defaultConfigThemeState: ConfigThemeState = {
+  expirationDate: undefined,
+  isSelected: false,
+};
+
+const calculateExpirationDate = (publishedAt?: string): Date => {
+  const baseDate = publishedAt ? new Date(publishedAt) : new Date();
+  return addDays(baseDate, CONFIG_THEME_EXPIRATION_DAYS);
+};
+
+const getOrCreateConfigThemeState = (
+  states: ConfigThemeStates,
+  uid: string,
+): ConfigThemeState => {
+  return states[uid] ?? { ...defaultConfigThemeState };
+};
+
+const initializeConfigThemeStates = (
+  configTheme: Partial<PartnerThemeConfig>,
+  persistedStates: ConfigThemeStates = {},
+): ConfigThemeStates => {
+  if (!configTheme.uid) {
+    return persistedStates;
+  }
+
+  const currentThemeUid = configTheme.uid;
+  const expirationDate = calculateExpirationDate(configTheme.publishedAt);
+
+  const existingState = persistedStates[currentThemeUid];
+  const existingExpirationDate = existingState?.expirationDate
+    ? new Date(existingState.expirationDate)
+    : undefined;
+
+  if (
+    existingState &&
+    (!configTheme.publishedAt ||
+      isEqual(existingExpirationDate, expirationDate))
+  ) {
+    return persistedStates;
+  }
+
+  return {
+    ...persistedStates,
+    [currentThemeUid]: {
+      expirationDate,
+      isSelected: true,
+    },
+  };
+};
 
 export const createThemeStore = (props: ThemeProps) =>
-  createWithEqualityFn<ThemeState>(
-    persist(
+  createWithEqualityFn(
+    persist<ThemeState, [], [], PersistedThemeState>(
       (set, get) => ({
         ...props,
         setConfigTheme: (configTheme: Partial<PartnerThemeConfig>) => {
-          set({
-            configTheme,
-          });
+          set({ configTheme });
         },
         setWidgetTheme: (widgetTheme: { config: Partial<WidgetConfig> }) => {
+          set({ widgetTheme });
+        },
+        setConfigThemeState: (
+          uid: string,
+          state: Partial<ConfigThemeState>,
+        ) => {
+          const currentStates = get().configThemeStates;
+          const currentState = getOrCreateConfigThemeState(currentStates, uid);
           set({
-            widgetTheme,
+            configThemeStates: {
+              ...currentStates,
+              [uid]: {
+                ...currentState,
+                ...state,
+              },
+            },
           });
         },
+        getConfigThemeState: (uid: string): ConfigThemeState => {
+          return getOrCreateConfigThemeState(get().configThemeStates, uid);
+        },
       }),
       {
         name: 'jumper-theme-store',
-        version: 1,
-        migrate: (persistedState: any, version: number) => {
+        version: 2,
+        storage: {
+          getItem: (name) => {
```

Comment:
I might be wrong but this should not be needed as it should be internally managed by zustand (getItem/setItem) within `      storage: createJSONStorage(() => sessionStorage), // (optional) by default, 'localStorage' is used
`

Reply:
Mmm the problem was that using `createJSONStorage` doesn't preserve the URL/Date objects and I wanted to avoid implementing our own serializer/deserializer